packages/devtools_app/lib/src/screens/inspector_v2/inspector_controller.dart

```diff
@@ -731,6 +731,25 @@ class InspectorController extends DisposableController
 
     try {
       final newSelection = await pendingSelectionFuture;
+
+      // Show an error and don't update the selected node in the tree if the
+      // user selected an implementation widget in the app while implementation
+      // widgets are hidden in the tree.
+      if (implementationWidgetsHidden.value && newSelection != null) {
+        final isInTree =
+            valueToInspectorTreeNode.containsKey(newSelection.valueRef);
+        final hasParent = newSelection.parent != null;
```

Comment:
just for my understanding, how does the presence of a parent tell us whether whether the user selected an implementation widget?

Reply:
The Widget Inspector's response includes the children for each node, but not the parent. When DevTools receives the response, it walks down the tree and sets the parent for a node. This is basically a way of check that not only is the node not in the current tree, but it was never part of the response we were given. Since we know the implementation widgets are hidden, we can assume that the selected node is an implementation widget.